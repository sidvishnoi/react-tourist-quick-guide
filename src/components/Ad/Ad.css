.Ad {
  background: #eee;
  margin: 0.5em auto;
  padding: 1em;
  text-align: center;
}
