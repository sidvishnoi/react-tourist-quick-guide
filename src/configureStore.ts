import { createStore, applyMiddleware, compose, StoreEnhancer } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { devToolsEnhancer } from 'redux-devtools-extension';
import rootReducer from './reducers';

export default function configureStore(preloadedState?: any) {
  const middlewares = [thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer, devToolsEnhancer({})];
  const composedEnhancers = compose(...enhancers) as StoreEnhancer<{}, {}>;

  const store = createStore(rootReducer, preloadedState, composedEnhancers);
  return store;
}
