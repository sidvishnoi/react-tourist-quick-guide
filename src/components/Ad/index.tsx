import * as React from 'react';
import './Ad.css';

export default function Ad() {
  return <div className="Ad">Advertisement Space</div>;
}
