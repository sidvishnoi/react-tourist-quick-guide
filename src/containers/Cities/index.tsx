import * as React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { removeCity } from '../../actions';
import Ad from '../../components/Ad';
import City, { CityProps } from '../../components/City';
import { State } from '../../state';

const mapStateToProps = (state: State) => ({
  cities: state.cities.map(name => ({
    distance: state.distance[name],
    name,
    places: state.places[name],
    weather: state.weather[name],
  })),
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    removeCity(name: string) {
      dispatch(removeCity(name));
    },
  };
};

interface Props {
  // from mapState
  cities: CityProps[];
  // from mapDispatch
  removeCity: (name: string) => void;
}

function Cities(props: Props) {
  return (
    <>
      {props.cities.slice(0, 1).map(city => (
        <City {...city} key={city.name} destroyer={props.removeCity} />
      ))}
      <Ad />
      {props.cities.slice(1).map(city => (
        <City {...city} key={city.name} destroyer={props.removeCity} />
      ))}
    </>
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Cities);
