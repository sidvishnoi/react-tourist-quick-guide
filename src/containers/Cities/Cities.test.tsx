import { mount, shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';

import Cities from '.';
import Ad from '../../components/Ad';
import configureStore from '../../configureStore';
import initialStateDefault, { State } from '../../state';

Enzyme.configure({ adapter: new Adapter() });

const initialState: State = {
  ...initialStateDefault,
  source: 'PASS',
  cities: {
    allIds: ['ONE', 'TWO'],
    byId: {
      ONE: {
        distance: {
          state: 'ready',
          data: 50,
        },
        weather: {
          state: 'ready',
          data: {
            icon: 113,
            summary: 'Sunny',
            temperature: 35,
            forecast: [
              { icon: 113, summary: 'Sunny', temperature: 35 },
              { icon: 113, summary: 'Sunny', temperature: 35 },
              { icon: 113, summary: 'Sunny', temperature: 35 },
            ],
          },
        },
        places: {
          state: 'ready',
          data: [
            { title: 'FOO', summary: 'foo', link: '#' },
            { title: 'BAR', summary: 'bar', link: '#' },
            { title: 'BAZ', summary: 'baz', link: '#' },
          ],
        },
      },
      TWO: {
        distance: { state: 'ready', data: 100 },
        weather: { state: 'error', data: null },
        places: { state: 'loading', data: null },
      },
    },
  },
};

describe('Containers - Cities', () => {
  it('renders cities when cities count is 0', () => {
    const store = configureStore({
      ...initialState,
      cities: {
        allIds: [],
        byId: {},
      },
    });
    const app = (
      <Provider store={store}>
        <Cities />
      </Provider>
    );

    const output = mount(app);
    expect(output.find('.city').length).toEqual(0);
    expect(output.containsMatchingElement(<Ad />)).toBeTruthy();

    const tree = renderer.create(app);
    expect(tree).toMatchSnapshot();
  });

  it('renders cities when cities count is more than 0', () => {
    const store = configureStore(initialState);
    const app = (
      <Provider store={store}>
        <Cities />
      </Provider>
    );

    const output = mount(app);

    expect(output.find('div.city').length).toEqual(2);
    expect(output.containsMatchingElement(<Ad />)).toBeTruthy();

    const tree = renderer.create(app);
    expect(tree).toMatchSnapshot();
  });

  it('allows removing city from list', () => {
    const store = configureStore(initialState);
    const app = (
      <Provider store={store}>
        <Cities />
      </Provider>
    );

    const output = mount(app);

    expect(output.find('div.city').length).toEqual(2);
    expect(store.getState().cities.allIds).toEqual(['ONE', 'TWO']);
    expect(renderer.create(app)).toMatchSnapshot();

    const removeButton = output.find('button.remove').first();
    removeButton.simulate('click');
    expect(output.find('div.city').length).toEqual(1);
    expect(store.getState().cities.allIds).toEqual(['TWO']);
    expect(renderer.create(app)).toMatchSnapshot();
  });

  it('allows moving a city up in list', () => {
    const store = configureStore(initialState);
    const app = (
      <Provider store={store}>
        <Cities />
      </Provider>
    );

    const output = mount(app);

    expect(output.find('div.city').length).toEqual(2);
    expect(store.getState().cities.allIds).toEqual(['ONE', 'TWO']);
    expect(renderer.create(app)).toMatchSnapshot();

    const removeButton = output.find('button.move').at(1);
    removeButton.simulate('click');
    expect(output.find('div.city').length).toEqual(2);
    expect(store.getState().cities.allIds).toEqual(['TWO', 'ONE']);
    expect(renderer.create(app)).toMatchSnapshot();
  });
});
