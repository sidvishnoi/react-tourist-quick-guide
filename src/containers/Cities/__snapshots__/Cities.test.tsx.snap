// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - Cities allows moving a city up in list 1`] = `
Array [
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        ONE
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        50 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div
        className="sc-iwsKbI pSYKQ"
      >
        <div
          className="weather-today"
        >
          <div
            className="weather-day sc-dnqmqq cJeYVJ"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
        <div
          className="weather-forecast sc-gZMcBi hguPkl"
        >
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div>
        <h3
          className="sc-htpNat hvSXnA"
          color="#000"
          size="fsize_05"
        >
          Top tourist places:
        </h3>
        <ul
          className="sc-htoDjs hEklmg"
        >
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                FOO
              </a>
              :
               
              <em>
                foo
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAR
              </a>
              :
               
              <em>
                bar
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAZ
              </a>
              :
               
              <em>
                baz
              </em>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div
    className="sc-bdVaJa hDTFWN"
  >
    Advertisement Space
  </div>,
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        TWO
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        100 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <p
        style={
          Object {
            "fontSize": "2em",
            "margin": 0,
          }
        }
      >
        ðŸ˜ž
      </p>
      <h4
        style={
          Object {
            "margin": 0,
          }
        }
      >
        Something went wrong.
      </h4>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <svg
        className="loader"
        height="50"
        stroke="#ff8a00"
        viewBox="0 0 38 38"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>,
]
`;

exports[`Containers - Cities allows moving a city up in list 2`] = `
Array [
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        TWO
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        100 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <p
        style={
          Object {
            "fontSize": "2em",
            "margin": 0,
          }
        }
      >
        ðŸ˜ž
      </p>
      <h4
        style={
          Object {
            "margin": 0,
          }
        }
      >
        Something went wrong.
      </h4>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <svg
        className="loader"
        height="50"
        stroke="#ff8a00"
        viewBox="0 0 38 38"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  <div
    className="sc-bdVaJa hDTFWN"
  >
    Advertisement Space
  </div>,
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        ONE
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        50 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div
        className="sc-iwsKbI pSYKQ"
      >
        <div
          className="weather-today"
        >
          <div
            className="weather-day sc-dnqmqq cJeYVJ"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
        <div
          className="weather-forecast sc-gZMcBi hguPkl"
        >
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div>
        <h3
          className="sc-htpNat hvSXnA"
          color="#000"
          size="fsize_05"
        >
          Top tourist places:
        </h3>
        <ul
          className="sc-htoDjs hEklmg"
        >
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                FOO
              </a>
              :
               
              <em>
                foo
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAR
              </a>
              :
               
              <em>
                bar
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAZ
              </a>
              :
               
              <em>
                baz
              </em>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;

exports[`Containers - Cities allows removing city from list 1`] = `
Array [
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        ONE
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        50 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div
        className="sc-iwsKbI pSYKQ"
      >
        <div
          className="weather-today"
        >
          <div
            className="weather-day sc-dnqmqq cJeYVJ"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
        <div
          className="weather-forecast sc-gZMcBi hguPkl"
        >
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div>
        <h3
          className="sc-htpNat hvSXnA"
          color="#000"
          size="fsize_05"
        >
          Top tourist places:
        </h3>
        <ul
          className="sc-htoDjs hEklmg"
        >
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                FOO
              </a>
              :
               
              <em>
                foo
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAR
              </a>
              :
               
              <em>
                bar
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAZ
              </a>
              :
               
              <em>
                baz
              </em>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div
    className="sc-bdVaJa hDTFWN"
  >
    Advertisement Space
  </div>,
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        TWO
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        100 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <p
        style={
          Object {
            "fontSize": "2em",
            "margin": 0,
          }
        }
      >
        ðŸ˜ž
      </p>
      <h4
        style={
          Object {
            "margin": 0,
          }
        }
      >
        Something went wrong.
      </h4>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <svg
        className="loader"
        height="50"
        stroke="#ff8a00"
        viewBox="0 0 38 38"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>,
]
`;

exports[`Containers - Cities allows removing city from list 2`] = `
Array [
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        TWO
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        100 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <p
        style={
          Object {
            "fontSize": "2em",
            "margin": 0,
          }
        }
      >
        ðŸ˜ž
      </p>
      <h4
        style={
          Object {
            "margin": 0,
          }
        }
      >
        Something went wrong.
      </h4>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <svg
        className="loader"
        height="50"
        stroke="#ff8a00"
        viewBox="0 0 38 38"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>,
  <div
    className="sc-bdVaJa hDTFWN"
  >
    Advertisement Space
  </div>,
]
`;

exports[`Containers - Cities renders cities when cities count is 0 1`] = `
<div
  className="sc-bdVaJa hDTFWN"
>
  Advertisement Space
</div>
`;

exports[`Containers - Cities renders cities when cities count is more than 0 1`] = `
Array [
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        ONE
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        50 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div
        className="sc-iwsKbI pSYKQ"
      >
        <div
          className="weather-today"
        >
          <div
            className="weather-day sc-dnqmqq cJeYVJ"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
        <div
          className="weather-forecast sc-gZMcBi hguPkl"
        >
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
          <div
            className="weather-day sc-dnqmqq cdKxNR"
            style={
              Object {
                "backgroundImage": "url(/images/weather-icons/113.png)",
              }
            }
          >
            <span>
              35
            </span>
            Â°C
            <p>
              Sunny
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-gqjmRU ebSWTF"
    >
      <div>
        <h3
          className="sc-htpNat hvSXnA"
          color="#000"
          size="fsize_05"
        >
          Top tourist places:
        </h3>
        <ul
          className="sc-htoDjs hEklmg"
        >
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                FOO
              </a>
              :
               
              <em>
                foo
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAR
              </a>
              :
               
              <em>
                bar
              </em>
            </span>
          </li>
          <li
            className="sc-gzVnrw iAYwO"
          >
            <span>
              <a
                className="sc-bZQynM ivuoxv"
                href="#"
              >
                BAZ
              </a>
              :
               
              <em>
                baz
              </em>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div
    className="sc-bdVaJa hDTFWN"
  >
    Advertisement Space
  </div>,
  <div
    className="city sc-jTzLTM gTeocR"
  >
    <div
      className="sc-bxivhb liRQLP"
    >
      <div
        className="buttons"
      >
        <button
          className="move sc-bwzfXH fGOwbX"
          color="#fff"
          onClick={[Function]}
          title="Move destination up in list"
        >
          â¬†
        </button>
        <button
          className="remove sc-bwzfXH gbuAGC"
          color="#fff"
          onClick={[Function]}
          title="Remove from list"
        >
          âœ—
        </button>
      </div>
      <h2
        className="sc-htpNat sc-ifAKCX jZwhHp"
        color="#fff"
        size="fsize_08"
      >
        TWO
      </h2>
      <span
        className="sc-htpNat sc-EHOje bUQOko"
        color="#fff"
        size="fsize_04"
      >
        100 KM
      </span>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <p
        style={
          Object {
            "fontSize": "2em",
            "margin": 0,
          }
        }
      >
        ðŸ˜ž
      </p>
      <h4
        style={
          Object {
            "margin": 0,
          }
        }
      >
        Something went wrong.
      </h4>
    </div>
    <div
      className="sc-gqjmRU sc-VigVT ewklmX"
    >
      <svg
        className="loader"
        height="50"
        stroke="#ff8a00"
        viewBox="0 0 38 38"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>,
]
`;
